WPFHospitalEditor/Building.xaml.cs

```diff
@@ -18,31 +23,73 @@ namespace WPFHospitalEditor
+        private List<MapObject> allBuildingObjects = new List<MapObject>();
 
-        public Building(int id)
+        MapObjectController mapObjectController = new MapObjectController(new MapObjectService(new MapObjectRepository(new FileRepository(AllConstants.MAPOBJECT_PATH))));
+
+        public Building(List<MapObject> buildingObjects, int selectedFloor)
         {
+            allBuildingObjects = buildingObjects;
             InitializeComponent();
-            clearAll();
-            if(id == 1)
+            clearAll();          
+            populateBuildingFloors(buildingObjects);            
+            setFloorComboBox();
+            floor.SelectedIndex = selectedFloor;            
+        }
+
+        private void populateBuildingFloors(List<MapObject> allBuildingObjects)
+        {          
+            int numberOfFloors = findMaxFloor(findAllFloors(allBuildingObjects));
+            for(int i = 0; i <= numberOfFloors; i++)
             {
-                buildingFloors.Add(0, new Floor(AllMapObjects.allFirstBuildingFirstFloorObjects));
-                buildingFloors.Add(1, new Floor(AllMapObjects.allFirstBuildingSecondFloorObjects));
-            } else
+                buildingFloors.Add(i, new Floor(findChosenFloor(allBuildingObjects,i)));
+            }
+        }
+
+        private List<int> findAllFloors(List<MapObject> allBuildingObjects)
+        {
+            List<int> floors = new List<int>();
+            foreach (MapObject mapObjectIteration in allBuildingObjects)
             {
-                buildingFloors.Add(0, new Floor(AllMapObjects.allSecondBuildingFirstFloorObjects));
-                buildingFloors.Add(1, new Floor(AllMapObjects.allSecondBuildingSecondFloorObjects));
+                floors.Add(int.Parse(findFloor(mapObjectIteration)));
             }
+            return floors;
+        }
 
-            hospitalMap.canvas = this.canvas;
-            setFloorComboBox();
-            hospitalMap.addObjectToCanvas(buildingFloors[0].getAllFloorMapObjects(), canvas);
-            displayLegend(buildingFloors[0].getAllFloorMapObjects());
-            floor.SelectedIndex = 0;
+        private int findMaxFloor(List<int> floors)
+        {
+            int maxFloor = 0;
+            for(int i = 0; i < floors.Count; i++)
+            {
+                if (floors[i] > maxFloor)
+                {
+                    maxFloor = floors[i];
+                }
+            }
+            return maxFloor;
+        }
+
+        private List<MapObject> findChosenFloor(List<MapObject> buildingObjects, int floor)
+        {
+            List<MapObject> floorObjects = new List<MapObject>();
+            foreach (MapObject mapObjectIteration in buildingObjects)
+            {
+                if (floor.ToString().Equals(findFloor(mapObjectIteration)))
+                {
+                    floorObjects.Add(mapObjectIteration);
+                }
+            }
+            return floorObjects;
         }
```

Comment:
It needs to be optimized as you go through the same list multiple times

Reply:
Done.